internal/iapl/policy.go

```diff
@@ -43,20 +47,29 @@ type Action struct {
 
 // ActionBinding represents a binding of an action to a resource type or union.
 type ActionBinding struct {
-	ActionName string
-	TypeName   string
-	Conditions []Condition
+	ActionName    string
+	TypeName      string
+	Conditions    []Condition
+	ConditionSets []types.ConditionSet
 }
 
 // Condition represents a necessary condition for performing an action.
 type Condition struct {
 	RoleBinding        *ConditionRoleBinding
+	RoleBindingV2      *ConditionRoleBindingV2
 	RelationshipAction *ConditionRelationshipAction
 }
 
 // ConditionRoleBinding represents a condition where a role binding is necessary to perform an action.
 type ConditionRoleBinding struct{}
 
+// ConditionRoleBindingV2 represents a condition where a role binding is necessary to perform an action.
+// Using this condition type in the policy will instruct the policy engine to
+// create all the necessary relationships in the schema to support RBAC V2
+type ConditionRoleBindingV2 struct {
+	InheritGrantsFrom []string
```

Comment:
Is there a reason to have this separate from `AvailableRolesFrom`? I can't really see a scenario where I would want to have roles available from one place, but inherit grants from another.